src/quipucordsctl/commands/uninstall.py

```diff
@@ -17,7 +20,158 @@ def get_help() -> str:
+                unit_file_config.read(unit_file_path)
+            except configparser.MissingSectionHeaderError:
+                logger.warning(
+                    _(
+                        "Skipping the %(unit_file)s container file due to"
+                        " missing section headers."
+                    ),
+                    {"unit_file": unit_file},
+                )
+
+            for section in unit_file_config.sections():
+                if section == "Container":
+                    if image := unit_file_config.get(section, "Image"):
+                        unique_images.add(image)
+
+    if not unique_images:
+        return True
+
+    all_removed = all(podman_utils.remove_image(image) for image in unique_images)
+    if not all_removed:
+        logger.warning(
+            _("At least one image failed to be removed"),
+        )
+    return True
+
+
+def remove_services() -> bool:
+    """Remove services."""
+    logger.info(
+        _("Removing the %(server_software_name)s services."),
+        {"server_software_name": settings.SERVER_SOFTWARE_NAME},
+    )
+
+    for env_file in settings.TEMPLATE_SERVER_ENV_FILENAMES:
+        env_file_path = Path(settings.SERVER_ENV_DIR) / env_file
+        if env_file_path.exists():
+            logger.info(
+                _("Removing the %(server_software_name)s env file %(env_file_path)s."),
+                {
+                    "server_software_name": settings.SERVER_SOFTWARE_NAME,
+                    "env_file_path": env_file_path,
+                },
+            )
+            env_file_path.unlink()
+
+    for unit_file in settings.TEMPLATE_SYSTEMD_UNITS_FILENAMES:
+        unit_file_path = Path(settings.SYSTEMD_UNITS_DIR) / unit_file
+        if unit_file_path.exists():
+            logger.info(
+                _(
+                    "Removing the %(server_software_name)s"
+                    " unit file %(unit_file_path)s."
+                ),
+                {
+                    "server_software_name": settings.SERVER_SOFTWARE_NAME,
+                    "unit_file_path": unit_file_path,
+                },
+            )
+            unit_file_path.unlink()
+
+    if (
+        settings.SYSTEMD_GENERATED_SERVICES_DIR
+        and settings.SYSTEMD_GENERATED_SERVICES_DIR.exists()
+    ):
+        for service_file in settings.SYSTEMD_SERVICE_FILENAMES:
+            service_file_path = settings.SYSTEMD_GENERATED_SERVICES_DIR / service_file
+            if service_file_path.exists():
+                logger.info(
+                    _(
+                        "Removing the %(server_software_name)s"
+                        " service file %(service_file_path)s."
+                    ),
+                    {
+                        "server_software_name": settings.SERVER_SOFTWARE_NAME,
+                        "service_file_path": service_file_path,
+                    },
+                )
+                service_file_path.unlink()
+    return True
```

Comment:
It looks like we _always_ `return True` here unless an exception is raised. I would remove the redundant return unless you have specific plans in motion for cases that would `return False`.

Maybe the guts of this function, or some of the inner ifs/for loops should be wrapped in their own try/except blocks to explicitly handle failures. For example, what happens if directory/file permissions unexpectedly prevent one of those `unlink()` calls from succeeding?